homeassistant/helpers/template.py

```diff
@@ -1183,6 +1183,12 @@ def is_device_attr(
     return bool(device_attr(hass, device_or_entity_id, attr_name) == attr_value)
 
 
+def areas(hass: HomeAssistant) -> Iterable[str | None]:
+    """Return all areas."""
+    area_reg = area_registry.async_get(hass)
+    return [area.id for area in area_reg.async_list_areas()]
```

Comment:
Why would it return area IDs and not area objects?

Reply:
I did follow the same logic as area_entities method

Reply:
I'm not sure, hence I question it to get a discussion going.

My main concern is thinking about `states` variable, I'm not sure if that is a confusing conflict/difference?

../Frenck

Reply:
Sure that with `Area` object it can be explored in more ways. 
I was thinking about listing all areas to be able to customize openai integration prompt, without additional code.
Right nor just returning ids we're able to retrieve all entities from each area.

Reply:
OpenAI already has `areas` available in the prompt template:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/openai_conversation/__init__.py#L152-L153

Reply:
I know that, but I think the call to retrieve areas could be inside the template.

Reply:
@frenck and @balloob as the docs were approved, can we end this discussion and merge?

Reply:
I've scheduled this PR for discussing in a Core meeting later today.

Reply:
Ok, we discussed it and think this is the right approach.